### PR TITLE
create enemies arrey, refsctor fireGatling to use array

### PR DESCRIPTION
**Pull request by Gehinom**
## Todo
- [x] [create enemies arrey, refsctor fireGatling to use array](https://github.com/DarthVanger/ghost-of-kyiv/pull/78/commits/1d23bb10cce683a7f9b94e20ea06954db7e40ea9)
- [x] update `moveEnemy` to use array
- [x] update `renderEnemy` to use array
- [x] update `collision` to use array
- [x] ...

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -24,38 +24,49 @@ let rocket = {
...
-  enemyHealth100.x = moveEnemy(enemyHealth100.x, enemyHealth100.velocity);
-  enemyHealth200.x = moveEnemy(enemyHealth200.x, enemyHealth200.velocity);
-  enemyHealth50text.x = moveEnemy(enemyHealth50text.x, enemyHealth50text.velocity);
-  enemyHealth100text.x = moveEnemy(enemyHealth100text.x, enemyHealth100text.velocity);
-  enemyHealth200text.x = moveEnemy(enemyHealth200text.x, enemyHealth200text.velocity);
+  //renderEnemy(enemyHealth50text, healthBar50text, enemyHealth50, healthBar50, enemy1);
+  //renderEnemy(enemyHealth100text, healthBar100text, enemyHealth100, enemyHealth100.element, enemy2);
+  //renderEnemy(enemyHealth200text, healthBar200text, enemyHealth200, healthBar200, enemy3);
+  // changed to
+  enemies.forEach(el => {renderEnemy(el)})
```
Comment (DarthVanger) [inline app.js:46]: Let's use a simpler version:
```
  enemies.forEach(function(el) {
    renderEnemy(el);
  })
  ```

File: app.js [hunk 2]
```diff
@@ -24,38 +24,49 @@ let rocket = {
...
+  //enemy1.x = moveEnemy(enemy1.x, enemy1.velocity);
+  //enemy2.x = moveEnemy(enemy2.x, enemy2.velocity);
+  //enemy3.x = moveEnemy(enemy3.x, enemy3.velocity);
+  //changed to
+  enemies.forEach(el => {moveEnemy({el})})
```
Comment (DarthVanger) [inline app.js:51]: You can simply write `enemines.forEach(moveEnemy);`

File: app.js [hunk 3]
```diff
@@ -24,38 +24,49 @@ let rocket = {
...
+  //enemyHealth50.x = moveEnemy(enemyHealth50.x, enemyHealth50.velocity);
+  //enemyHealth100.x = moveEnemy(enemyHealth100.x, enemyHealth100.velocity);
+  //enemyHealth200.x = moveEnemy(enemyHealth200.x, enemyHealth200.velocity);
+  //changed to 
+  enemies.forEach(el => {el})
```
Comment (DarthVanger) [inline app.js:56]: This does nothing, remove this line

File: app.js [hunk 4]
```diff
@@ -24,38 +24,49 @@ let rocket = {
...
+
   moveRocket();
-  checkEnemyShipCollision(enemy1);
-  checkEnemyShipCollision(enemy2);
-  checkEnemyShipCollision(enemy3);
+  //checkEnemyShipCollision(enemy1);
+  //checkEnemyShipCollision(enemy2);
+  //checkEnemyShipCollision(enemy3);
+  //changed to
+  enemies.forEach(el => {checkEnemyShipCollision(el)})
```
Comment (DarthVanger) [inline app.js:67]: Simplify to `enemies.forEach(checkEnemyShipCollision);`

File: app.js [hunk 5]
```diff
@@ -104,7 +115,29 @@ function Step() {
       rocket.dmg = 50;
       rocket.element.src = "img/Rocket.gif";
     }
-  }
+  }*/
+  //changed to
+  enemies.forEach((el) => {
```
Comment (DarthVanger) [inline app.js:120]: Lett's create a function, so we have `enemies.forEach(collisionSHmolision)`;

```
enemies.forEach(collisionSHmolision);

function collisionSHmolision(el) {
 if(checkEnemyRocketCollision(el)){
      el.enemyHealth.element.value -= rocket.dmg;
      rocket.dmg -= rocket.dmg;
      soundRocketHit.play();
      }
      if (el.enemyHealth.element.value <= 0 && el.isOn) {
        el.x -= enemyDies;
        soundEnemyDieExplosion.play();
        el.isOn = false;
      }
      if (rocket.dmg <= 0) {
        rocket.x = airfighter.x + airfighter.rocketDefaultX;
        rocket.y = airfighter.y + airfighter.rocketDefaultY;
        rocket.velocity -= 8;
        rocket.dmg = 50;
        rocket.element.src = "img/Rocket.gif";
      }
    }
}

File: app.js (hunk 2 above)
Comment (DarthVanger) [reply app.js:51]: Secondly, you are passing `el` inside and object: `{el}`.

And you are receiving it as `{el}` object in the `moveEnemy` function as well.

Change `function moveEnemy({el}) {` to `function moveEnemy(el) {`.
And use ` enemines.forEach(moveEnemy);`.

This should have the same effect as your code, but with less symbols :)

File: enemy.js [hunk 6]
```diff
@@ -1,26 +1,39 @@
-export let enemy1 = {
+import {enemyHealth50, enemyHealth100, enemyHealth200, enemyHealth50text, enemyHealth100text, enemyHealth200text} from "./health.js";
+
+let enemy1 = {
+    isOn : true,
```
Comment (DarthVanger) [inline enemy.js:4]: Please rename `isOn` -> `isKilled` or `isOn` -> `isAlive`.

**Review by DarthVanger — APPROVED** [review]
image

**Review by 19thanks96 — APPROVED** [review]
LGTM

**Pull request merged**